app/app.go

```diff
@@ -0,0 +1,199 @@
+	"io"
+	"strings"
+
+	"github.com/datatrails/go-datatrails-logverification/logverification/app"
+)
+
+/**
+ * Merklelog related APP contents
+ *
+ * An APP in the context of merklelog is an interface that commits log entries.
+ *
+ * Apps include:
+ *  - assetsv2
+ *  - eventsv1
+ */
+
+const (
+	AssetsV2AppDomain = byte(0)
+	EventsV1AppDomain = byte(1)
+)
+
+var (
+	ErrNoJsonGiven = errors.New("no json given")
+)
+
+// AppDataToVerifiableLogEntries converts the app data to verifiable log entries
+func AppDataToVerifiableLogEntries(appData []byte, logTenant string) ([]app.AppEntry, error) {
+
+	// first attempt to convert the appdata to a list of events
+	eventList, err := eventListFromData(appData)
+	if err != nil {
+		return nil, err
+	}
+
+	// now we have an event list we can decipher if the app is
+	//  assetsv2 or eventsv1
+	appDomain := appDomain(appData)
+
+	verifiableLogEntries := []app.AppEntry{}
+
+	switch appDomain {
+	case AssetsV2AppDomain:
+		// assetsv2
+		verfiableAssetsV2Events, err := NewAssetsV2AppEntries(eventList)
+		if err != nil {
+			return nil, err
+		}
+
+		verifiableLogEntries = append(verifiableLogEntries, verfiableAssetsV2Events...)
+
+	case EventsV1AppDomain:
+		verfiableEventsV1Events, err := NewEventsV1AppEntries(eventList, logTenant)
+		if err != nil {
+			return nil, err
+		}
+
+		verifiableLogEntries = append(verifiableLogEntries, verfiableEventsV1Events...)
+
+	default:
+		return nil, errors.New("unknown app domain for given app data")
+	}
+
+	return verifiableLogEntries, nil
+}
+
+// appDomain returns the app domain of the given app data
+func appDomain(appData []byte) byte {
+
+	// first attempt to convert the appdata to a list of events
+	eventList, err := eventListFromData(appData)
+	if err != nil {
+		// if we can't return default of assetsv2
+		return AssetsV2AppDomain
+	}
+
+	// decode into events
+	events := struct {
+		Events        []json.RawMessage `json:"events,omitempty"`
+		NextPageToken json.RawMessage   `json:"next_page_token,omitempty"`
+	}{}
+
+	decoder := json.NewDecoder(bytes.NewReader(eventList))
+	decoder.DisallowUnknownFields()
+	for {
+		err = decoder.Decode(&events)
+
+		if errors.Is(err, io.EOF) {
+			break
+		}
+
+		if err != nil {
+			// return default of assetsv2
+			return AssetsV2AppDomain
+		}
+	}
+
+	// decode the first event and find the identity
+	event := struct {
+		Identity string `json:"identity,omitempty"`
+	}{}
+
+	decoder = json.NewDecoder(bytes.NewReader(events.Events[0]))
+	decoder.DisallowUnknownFields()
+
+	for {
+		err = decoder.Decode(&event)
+
+		if errors.Is(err, io.EOF) {
+			break
+		}
+
+		if err != nil {
+			// if we can't return default of assetsv2
+			return AssetsV2AppDomain
+		}
+	}
+
+	// find if the event identity is assetsv2 or eventsv1 identity
+	if strings.HasPrefix(event.Identity, "assets/") {
+		return AssetsV2AppDomain
+	} else {
+		return EventsV1AppDomain
+	}
+
+}
+
+// eventListFromData normalises a json encoded event or *list* of events, by
```

Comment:
Minor: Lets rename data to something more descriptive, as it doesn't help the reader understand what type of data to expect

Reply:
we should come up with a decent term for it then, 

currently in our term sheet data is used for many entries for something.

i.e. mmrData => many mmr Entries...
trieData => many trie Entries
appData => many app Entries

Reply:
i agree data is very generic and not very descriptive :)

Reply:
The examples you listed do tell us a little more though (its related to the MMR itself, the trie we construct or the application.) In this case, the data we're describing is an event list API response.

Reply:
thats very true :) will rename

Reply:
renamed to eventListFromJson :) as its json we expect as input

Reply:
Its hideously accurate, I like it <3